packages/ra-ui-materialui/src/list/DatagridHeaderCell.js

```diff
@@ -23,17 +23,18 @@ const useStyles = makeStyles(
     { name: 'RaDatagridHeaderCell' }
 );
 
-export const DatagridHeaderCell = ({
-    className,
-    classes: classesOverride,
-    field,
-    currentSort,
-    updateSort,
-    resource,
-    isSorting,
-    ...rest
-}) => {
-    const classes = useStyles({ classes: classesOverride });
+export const DatagridHeaderCell = props => {
+    const {
+        className,
+        classes: classesOverride,
```

Comment:
You probably don't want to destructure this one or it won't be passed to `useStyles` preventing overrides

Reply:
I'm passing `props` to `useStyles`, not `rest`. Destructuring as `classesOverride` here avoids the name conflict with the `useStyles` return value.

Reply:
Read too quickly...